x/config/doc.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package config implements a generic configuration system that may be used
+// to build YARPC Dispatchers from configurations specified in different
+// markup formats.
+//
+// Usage
+//
+// To build a Dispatcher, set up a Loader and inform it about the different
+// transports that it needs to support. Use Load or LoadYAML to load a Builder
+// from the given configuration.
+//
+// 	loader := config.NewLoader()
+// 	http.RegisterTransport(loader)
+// 	redis.RegisterTransport(loader)
+//
+// 	builder, err := loader.LoadYAML(yamlConfig)
```

Comment:
Naming is still in flux but we do want the separate type simply for logging/debugging purposes. Pretty printing a builder will give you the exact configuration you're about to build your Dispatcher on. It'll be more informative than logging your `yarpc.Config` object which doesn't have a very readable pretty-printed representation because it has references to HTTP/TChannel clients, etc.